src/main/cpp/chassis/headingStates/MaintainHeading.cpp

```diff
@@ -28,29 +28,24 @@ MaintainHeading::MaintainHeading() : ISwerveDriveOrientation(ChassisOptionEnums:
 
 void MaintainHeading::UpdateChassisSpeeds(ChassisMovement &chassisMovement)
 {
-    units::angular_velocity::degrees_per_second_t correction = units::angular_velocity::degrees_per_second_t(0.0);
-
-    units::radians_per_second_t rot = chassisMovement.chassisSpeeds.omega;
     auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
     auto chassis = config != nullptr ? config->GetSwerveChassis() : nullptr;
-
-    if (units::math::abs(rot).to<double>() > 0.1)
-    {
-        chassis->SetStoredHeading(chassis->GetPose().Rotation().Degrees());
-    }
-    else
+    if (chassis != nullptr)
     {
-        chassisMovement.chassisSpeeds.omega = units::radians_per_second_t(0.0);
-        double error = abs(chassis->GetPose().Rotation().Degrees().to<double>() - chassis->GetStoredHeading().to<double>());
-        if (error < 5.0)
-            correction = CalcHeadingCorrection(chassis->GetStoredHeading(), m_kPMaintainFine);
-        else
-            correction = CalcHeadingCorrection(chassis->GetStoredHeading(), m_kPMaintainCoarse);
+        auto correction = units::angular_velocity::degrees_per_second_t(0.0);
+
+        auto translatingOrStrafing = (abs(chassisMovement.chassisSpeeds.vx.to<double>()) +
+                                      abs(chassisMovement.chassisSpeeds.vy.to<double>())) > 0.0;
+        if (abs(chassisMovement.chassisSpeeds.omega.to<double>()) > 0.0 || !m_prevTranslatinOrStrafing)
```

Comment:
I don't see m_prevTranslatinOrStrafing being set any where? What is the purpose of this?

Reply:
Inadvertent delete when cleaning up.  Put it back.  Thx